### PR TITLE
fix(ui): use hooks for meta/env/user

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes NAN-558
Contributes to NAN-516

We were spamming a bit /meta /env /user because we were using straight http call in the components. 
This is tentative to start refactoring those calls using SWR and hooks. I also need to add stuff to `/meta` for the new interactive demo and it was a good time to have something stable to do it later.

- Start exporting some types from the Server
We are duplicating types from server to webapp, which creates discrepancy. Start small by exposing `GetMeta` to see if the forms is fitting.

- Introduce `useEnvironment` `useMeta` `useUser` 
This should remove the biggest offender that fetch on every pages. This also reduce the boilerplate needed on some pages, but could be reduced further by removing some useless setState.

### CODE REVIEW

**Comment by bodinsamuel** [comment]
~There is a weird things happening to the dom, let me check that before reopening~ nevermind it was my browser cache

File: packages/webapp/src/App.tsx [hunk 1]
```diff
@@ -100,6 +100,11 @@ const App = () => {
         >
             <SWRConfig
                 value={{
+                    refreshInterval: 15 * 60000,
```
Comment (khaliqgant) [inline packages/webapp/src/App.tsx:103]: Can this be overridden on the activity page to a shorter value? It has been quite nice to have refreshes on that page . Temporal does a 15 second refresh FWIW

image

**Comment by khaliqgant** [comment]
Clicking through the nav and the activity page just doesn't load:


https://github.com/NangoHQ/nango/assets/1724137/d83e33b4-26c0-4947-9faa-15b993de4d03

File: packages/webapp/package.json [hunk 2]
```diff
@@ -33,6 +33,7 @@
         "@heroicons/react": "2.0.18",
         "@mantine/prism": "5.10.5",
         "@nangohq/frontend": "^0.39.3",
+        "@nangohq/server": "file:../server",
```
Comment (khaliqgant) [inline packages/webapp/package.json:36]: nice

File: packages/webapp/src/App.tsx (hunk 1 above)
Comment (bodinsamuel) [reply packages/webapp/src/App.tsx:103]: it's configurable at the call level yes. Activity page is not using SWR yet no?

Comment (bodinsamuel) [reply packages/webapp/src/App.tsx:103]: nevermind it does, changing that 👍🏻

Comment (khaliqgant) [reply packages/webapp/src/App.tsx:103]: Thanks. That page was the entire reason I added in SWR last minute FWIW

**Comment by bodinsamuel** [comment]
> Clicking through the nav and the activity page just doesn't load:

Indeed, thanks for raising, there was an issue with the fetcher cache, it was not triggering `onSuccess` so I had to change the logic to something more hooky. Let me know

**Comment by bodinsamuel** [comment]
@khaliqgant it's in staging

**Pull request merged**